apier/v1/attributes.go

```diff
@@ -59,6 +59,16 @@ func (apierV1 *ApierV1) SetAttributeProfile(alsPrf *engine.AttributeProfile, rep
 	if missing := utils.MissingStructFields(alsPrf, []string{"Tenant", "ID"}); len(missing) != 0 {
 		return utils.NewErrMandatoryIeMissing(missing...)
 	}
+	if len(alsPrf.Attributes) != 0 {
+		for _, attr := range alsPrf.Attributes {
+			for _, sub := range attr.Substitute {
+				if missing := utils.MissingStructFields(sub, []string{"Rules", "AllFiltersMatch"}); len(missing) != 0 {
```

Comment:
Please make AllFiltersMatch as optional (ie: if attr.Substitute.Rules == ""...

Reply:
Also, I would compile the rule before saving it since there are chances that the rule is wrongly inserted, hence we will inject wrong code in db.